database/migration/src/main/resources/changesets/enlarge_job_store_ids.xml

```diff
@@ -0,0 +1,108 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<databaseChangeLog objectQuotingStrategy="QUOTE_ALL_OBJECTS"
+                   xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+                   xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                   xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.3.xsd">
+
+    <!-- Drop the foreign key constraint from JOB_STORE_SIMPLETON_ENTRY to JOB_STORE_ENTRY to allow for the latter's PK
+         to be widened. -->
+    <changeSet author="mcovarr" id="drop_job_store_simpleton_entry_fk" dbms="mysql,hsqldb,postgresql,mariadb">
+        <dropForeignKeyConstraint
+                baseTableName="JOB_STORE_SIMPLETON_ENTRY"
+                constraintName="FK_JOB_STORE_SIMPLETON_ENTRY_JOB_STORE_ENTRY_ID"
+        />
+    </changeSet>
+
+    <!-- BEGIN JOB_STORE_ENTRY PK widening -->
+    <changeSet author="mcovarr" id="enlarge_job_store_entry_id" dbms="mysql,hsqldb,mariadb">
+        <addAutoIncrement
+                columnName="JOB_STORE_ENTRY_ID"
+                columnDataType="BIGINT"
+                incrementBy="1"
+                tableName="JOB_STORE_ENTRY"
+        />
+    </changeSet>
+
+    <changeSet author="mcovarr" id="postgresql_enlarge_job_store_entry_id" dbms="postgresql">
+        <modifyDataType
+                columnName="JOB_STORE_ENTRY_ID"
+                tableName="JOB_STORE_ENTRY"
+                newDataType="BIGINT"
+        />
+    </changeSet>
+
+    <changeSet author="mcovarr" id="postgresql_enlarge_job_store_entry_id_seq" dbms="postgresql">
+        <preConditions onFail="MARK_RAN">
+            <sqlCheck expectedResult="0">
+                SELECT count(*)
+                FROM information_schema.sequences
+                WHERE sequence_name = 'JOB_STORE_ENTRY_JOB_STORE_ENTRY_ID_seq'
+                AND data_type = 'bigint';
+            </sqlCheck>
+        </preConditions>
+        <sql>alter sequence "JOB_STORE_ENTRY_JOB_STORE_ENTRY_ID_seq" as bigint;</sql>
+    </changeSet>
```

Comment:
Could you comment why this precondition check is needed?

Reply:
I "borrowed" this from Khalid's earlier pg work but this appears to be an idempotency check (noop if the sequence is present and already consistent what the `alter` would do).